simulator/func_sim/rf/rf.h

```diff
@@ -79,13 +84,29 @@ class RF
         else if ( accumulating_instr == -1)
             val = read_hi_lo() - val;
 
+        if( loadlr)
+        {
+            if(loadlr > 0)
+            {
+                uint32 reg_val = static_cast<uint32>(get_value( num));
```

Comment:
Please add some comments in that block. I understand what happens, but it took some time.